src/Compilers/CSharp/Portable/Symbols/Source/SourceParameterSymbol.cs

```diff
@@ -74,7 +74,8 @@ internal abstract class SourceParameterSymbol : SourceParameterSymbolBase
                 !isExtensionMethodThis &&
                 (syntax.Default == null) &&
                 (syntax.AttributeLists.Count == 0) &&
-                !owner.IsPartialMethod())
+                !owner.IsPartialMethod() &&
+                !isRecordPrimaryConstructorParameter(syntax))
```

Comment:
don't know if you want docs here.  but we don't want want to waste the space here for simple parameters.  so if we're a record-primary-constructor parameter, we make sure to create a SourceComplexParameterSymbol